src/Workspaces/Core/Portable/NamingStyles/Serialization/NamingStylePreferences.cs

```diff
@@ -62,11 +63,7 @@ internal XElement CreateXElement()
 
         internal static NamingStylePreferences FromXElement(XElement element)
         {
-            var serializationVersion = int.Parse(element.Attribute("SerializationVersion").Value);
-            if (serializationVersion != s_serializationVersion)
-            {
-                element = XElement.Parse(DefaultNamingPreferencesString);
-            }
+            element = GetUpgradedSerializationIfNecessary(element);
```

Comment:
tagging @sharwell @jcouv .  Can this be discussed at the design meeting?  This may be the first case where we change serialized settings, but want to ensure that old settings still work.  We should decide on what this means and what back/forward-compat invariants we provide.  For example, what happens if a person uses both vs.old and vs.new in a team with shared settings, and if that behavior is ok.

Reply:
i don't think this is the first case... the current version number is 4, so it must have been increased 3 times already?

Reply:
i don't believe so.  my recollection is that we shipped at v4.  we've never had to change the setting since hten.  you're the first, and so this is the time that we need to figure out hte compat story.